src/rgw/rgw_rest.cc

```diff
@@ -673,7 +673,7 @@ void end_header(struct req_state* s, RGWOp* op, const char *content_type,
   /* do not send content type if content length is zero
      and the content type was not set by the user */
   if (force_content_type ||
-      (!content_type &&  s->formatter->get_len()  != 0) || s->err.is_err()){
```

Comment:
#0  end_header (s=0x7f7f917f8160, op=0x7f7db4017030, content_type=0x0, proposed_content_length=-1, force_content_type=false, force_no_error=false) at rgw/rgw_rest.cc:637
#1  RGWRESTFlusher::do_start (this=0x7f7db40170c0, ret=0) at rgw/rgw_rest.cc:997
#2  RGWFormatterFlusher::start (this=0x7f7db40170c0, client_ret=0) at rgw/rgw_formats.h:82
#3  RGWUserAdminOp_User::create (store=0x7f824d275240, op_state=..., flusher=...) at rgw/rgw_user.cc:2326
#4  RGWOp_User_Create::execute (this=0x7f7db4017030) at rgw/rgw_rest_user.cc:146
#5  process_request (store=0x7f824d275240, rest=0x7fffa583ec20, req=0x7f7f80015c60, client_io=0x7f7f917f8750, olog=0x0) at rgw/rgw_process.cc:270
#6  RGWFCGXProcess::handle_request (this=0x7f824d3515d0, r=0x7f7f80015c60) at rgw/rgw_fcgi_process.cc:121
#7  RGWProcess::RGWWQ::_process (this=0x7f824d351888, req=0x7f7f80015c60) at rgw/rgw_process.h:85
#8  ThreadPool::WorkQueue<RGWRequest>::_void_process (this=0x7f824d351888, p=0x7f7f80015c60, handle=...) at ./common/WorkQueue.h:282
#9  ThreadPool::worker (this=0x7f824d351640, wt=0x7f7f800008f0) at common/WorkQueue.cc:128
#10 ThreadPool::WorkThread::entry (this=0x7f7f800008f0) at common/WorkQueue.h:445
#11 Thread::entry_wrapper (this=0x7f7f800008f0) at common/Thread.cc:87
#12 Thread::_entry_func (arg=0x7f7f800008f0) at common/Thread.cc:67
#13 start_thread (arg=0x7f7f917fa700) at pthread_create.c:312